File: src/main/java/net/ndrei/teslacorelib/inventory/FluidTank.java
package net.ndrei.teslacorelib.inventory;

import net.minecraft.nbt.NBTTagCompound;

/**
 * Created by CF on 2016-12-29.
 */
public class FluidTank extends net.minecraftforge.fluids.FluidTank implements ISerializableFluidTank {
    public FluidTank(int capacity) {
        super(capacity);
    }

    @Override
    public NBTTagCompound serializeNBT() {
        NBTTagCompound nbt = new NBTTagCompound();
        return super.writeToNBT(nbt);
    }

    @Override
    public void deserializeNBT(NBTTagCompound nbt) {
        super.readFromNBT(nbt);
    }
}

File: src/main/java/net/ndrei/teslacorelib/inventory/FluidTank.kt
package net.ndrei.teslacorelib.inventory

import net.minecraft.nbt.NBTTagCompound

/**
 * Created by CF on 2017-06-28.
 */
open class FluidTank(capacity: Int)
    : net.minecraftforge.fluids.FluidTank(capacity), ISerializableFluidTank {

    override fun serializeNBT(): NBTTagCompound {
        val nbt = NBTTagCompound()
        return super.writeToNBT(nbt)
    }

    override fun deserializeNBT(nbt: NBTTagCompound) {
        super.readFromNBT(nbt)
    }
}
